hpat/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2218,3 +2218,136 @@ def hpat_pandas_series_median_impl(self, axis=None, skipna=True, level=None, num
         return numpy.median(self._data)
 
     return hpat_pandas_series_median_impl
+
+
+@overload_method(SeriesType, 'fillna')
+def hpat_pandas_series_fillna(self, value=None, method=None, axis=None, inplace=False, limit=None, downcast=None):
+    """
+    Pandas Series method :meth:`pandas.Series.fillna` implementation.
+
+    .. only:: developer
+
+       Tests: python -m hpat.runtests -k hpat.tests.test_series.TestSeries.test_series_fillna*
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        input series
+    value: scalar of the same dtype as input Series (other types currently unsupported), default None
+        Value to use to fill the NA elements
+    method: :obj:`string` {`backfill`, `bfill`, `pad`, `ffill`, None}, default None
+        Method to use for filling holes in reindexed Series.
+        *unsupported*
+    axis: :obj:`int` or :obj:`string` {0 or `index`}, default None
+        There is only one axis to drop values from.
+    inplace: :obj:`bool`, default False
+        If True, do operation inplace and return None.
+        Supported as literal value only
+    limit: :obj:`int`, default None
+        If method is specified, this is the maximum number of consecutive NaN
+        values to forward/backward fill.
+        *unsupported*
+    downcast: :obj:`dict` or :obj:`string` {`infer`}, default None
+        Controls logic of downcasting elements to particular dtype
+        *unsupported*
+    Returns
+    -------
+    :obj:`pandas.Series`
+         returns :obj:`pandas.Series` with missed values filled.
+    """
+
+    _func_name = 'Method fillna().'
+
+    if not isinstance(self, SeriesType):
+        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+
+    if not (isinstance(axis, (types.Integer, types.UnicodeType, types.Omitted)) or axis is None):
+        raise TypingError('{} The axis must be an Integer or String. Given: {}'.format(_func_name, axis))
+
+    if not (isinstance(inplace, types.Literal) and isinstance(inplace, types.Boolean)
+            or isinstance(inplace, types.Omitted)
+            or inplace == False):
+        raise TypingError('{} The inplace must be a literal Boolean constant. Given: {}'.format(_func_name, inplace))
+
+    if not ((method is None or isinstance(method, types.Omitted))
+            and (limit is None or isinstance(limit, types.Omitted))
+            and (downcast is None or isinstance(downcast, types.Omitted))
+    ):
```

Comment:
What do you think about splitting this condition on 3 independent ones? At the least it would look simpler.

Reply:
@densmirn I don't really think it's better, cause we'll duplicate raise statement, but I agree the check can be simplified if we, e.g. use a function like this:
```
def is_arg_omitted(arg, default_value):

    has_default_value = (arg is default_value) if (default_value is None) else (arg == default_value)
    return has_default_value or isinstance(arg, types.Omitted)
```

Can you maybe incorporate it into your TypeChecker in #241?

Reply:
We can update `TypeChecker` with the change, but with other commit (maybe with this one) and after `TypeChecker` pushed to master.